packages/hash/api/src/graphql/resolvers/knowledge/entity/before-update-entity-hooks.ts

```diff
@@ -77,6 +77,17 @@ const userEntityHookCallback: BeforeUpdateEntityHookCallback = async ({
       throw new ApolloError("Cannot unset preferred name");
     }
   }
+
+  const currentEmails = userModel.getEmails();
+
+  const updatedEmails: string[] =
+    updatedProperties[SYSTEM_TYPES.propertyType.email.baseUri];
+
+  if (currentEmails.sort().join() !== updatedEmails.sort().join()) {
```

Comment:
Sort here would modify the emails in-place. I don't think this matters too much, but it would mean that the kratos trait containing the list of emails would sometimes be a different order than the email list on the user entity

Reply:
That's something I hadn't considered, thanks for flagging. Although I don't think it would've caused any bugs, it seems better practice to not be modifying these arrays without purpose. Have addressed this in https://github.com/hashintel/hash/pull/1364/commits/1ca9712a4421d0ac367085b9422edf4b58d84473